src/plugins/customBorders/customBorders.js

```diff
@@ -364,25 +363,169 @@ class CustomBorders extends BasePlugin {
    * @param {Array} customBorders Object with `row` and `col`, `left`, `right`, `top` and `bottom` properties.
    */
   createCustomBorders(customBorders) {
-    rangeEach(0, customBorders.length - 1, (index) => {
-      if (customBorders[index].range) {
-        this.prepareBorderFromCustomAddedRange(customBorders[index]);
+    arrayEach(customBorders, (customBorder) => {
+      if (customBorder.range) {
+        this.prepareBorderFromCustomAddedRange(customBorder);
 
       } else {
-        this.prepareBorderFromCustomAdded(customBorders[index].row, customBorders[index].col, customBorders[index]);
+        this.prepareBorderFromCustomAdded(customBorder.row, customBorder.col, customBorder);
       }
     });
+  }
+
+  /**
+  * Clear borders settings from custom selections.
+  *
+  * @private
+  * @param {String} borderId Border id name as string.
+  */
+  clearBordersFromSelectionSettings(borderId) {
+    const index = arrayMap(this.hot.selection.highlight.customSelections, (customSelection) => customSelection.settings.id).indexOf(borderId);
+
+    if (index > -1) {
+      this.hot.selection.highlight.customSelections[index].clear();
+    }
 
-    this.hot.render();
     this.hot.view.wt.draw(true);
   }
 
   /**
-   * Add border options to context menu.
+  * Clear cellRange with null value.
+  *
+  * @private
+  */
+  clearNullCellRange() {
+    arrayEach(this.hot.selection.highlight.customSelections, (customSelection, index) => {
+      if (customSelection.cellRange === null) {
+        this.hot.selection.highlight.customSelections.splice(index, 1);
+        return false; // breaks forAll
+      }
+    });
+  }
+
+  /**
+    * Clear borders.
+    *
+    * @private
+    */
+  clearBorders() {
+    arrayEach(this.savedBorders, (border) => {
+      let borderId = border.id;
+
+      this.clearBordersFromSelectionSettings(borderId);
+    });
+  }
+
+  /**
+  * Splice border from savedBorders.
+  *
+  * @private
+  * @param {String} borderId Border id name as string.
+  */
+  spliceBorder(borderId) {
+    const index = arrayMap(this.savedBorders, (border) => border.id).indexOf(borderId);
+
+    if (index > -1) {
+      this.savedBorders.splice(index, 1);
+    }
+  }
+
+  /**
+  * Check if an border already exists in the savedBorders array, and if true update border in savedBorders.
+  *
+  * @private
+  * @param {Object} border Object with `row` and `col`, `left`, `right`, `top` and `bottom`, `id` and `border` ({Object} with `color`, `width` and `cornerVisible` property) properties.
+  *
+  * @return {Boolean}
+  */
+  checkSavedBorders(border) {
+    let check = false;
+
+    arrayEach(this.savedBorders, (savedBorder, index) => {
+      if (border.id === savedBorder.id) {
+        this.savedBorders[index] = border;
+
+        check = true;
```

Comment:
Here you can add `return false` statement to stop the loop after detected the border collision.

Reply:
Done.